src/diracx/db/jobs/status_utility.py

```diff
@@ -146,3 +150,195 @@ async def set_job_status(
         )
 
     return SetJobStatusReturn(**job_data)
+
+
+async def delete_job(
+    job_id: int,
+    config: Config,
+    job_db: JobDB,
+    job_logging_db: JobLoggingDB,
+    task_queue_db: TaskQueueDB,
+    background_task: BackgroundTasks,
+):
+    """
+    Delete a job by killing and setting the job status to DELETED.
+    """
+
+    await __kill_delete_job(
+        job_id,
+        config,
+        job_db,
+        task_queue_db,
+        background_task,
+    )
+
+    await set_job_status(
+        job_id,
+        {
+            datetime.now(timezone.utc): JobStatusUpdate(
+                Status=JobStatus.DELETED,
+                MinorStatus="Checking accounting",
+                StatusSource="job_manager",
+            )
+        },
+        job_db,
+        job_logging_db,
+        force=True,
+    )
+
+
+async def kill_job(
+    job_id: int,
+    config: Config,
+    job_db: JobDB,
+    job_logging_db: JobLoggingDB,
+    task_queue_db: TaskQueueDB,
+    background_task: BackgroundTasks,
+):
+    await __kill_delete_job(
```

Comment:
I'd avoid double underscore method names as they provide very little value most of the time:
```suggestion
    await _kill_delete_job(
```